canvas_api_client/canvas_api_client/interface.py

```diff
@@ -42,6 +42,22 @@ def delete_enrollment(self,
         """
         pass
 
+    @abstractmethod
+    def put_page(self,
+                 course_id: str,
+                 body: str,
+                 is_sis_course_id: bool = False,
+                 url: str = None,
+                 title: str = None,
```

Comment:
Since title is a required field for course pages, Perhaps this should not be an optional field?
```
columbiasce.test.instructure.com/api/v1/courses/59353/pages/None
```
Also, a nice thing to do for the other fields that are optional is to add the `Optional` type hint both here and in the v1_client.py file. For example: `url: Optional[str] = None,` -- this allows users to read the code more easily as well as type check the code with mypy.

Reply:
Okay, but title is not a required field; you can create a page with only a body